pkg/controller/operands/namespace.go

```diff
@@ -0,0 +1,103 @@
+package operands
+
+import (
+	"errors"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/pkg/apis/hco/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/controller/common"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+type namespaceHandler genericOperand
+
+func newNamespaceHandler(Client client.Client, Scheme *runtime.Scheme) *namespaceHandler {
+	return &namespaceHandler{
+		Client:                 Client,
+		Scheme:                 Scheme,
+		crType:                 "Namespace",
+		removeExistingOwner:    false,
+		setControllerReference: false,
+		skipCreate:             true,
+		hooks:                  &namespaceHooks{},
+	}
+}
+
+type namespaceHooks struct{}
+
+func (h namespaceHooks) getFullCr(hc *hcov1beta1.HyperConverged) (client.Object, error) {
+	return NewNamespace(hc), nil
+}
+func (h namespaceHooks) getEmptyCr() client.Object { return &corev1.Namespace{} }
+func (h namespaceHooks) getObjectMeta(cr runtime.Object) *metav1.ObjectMeta {
+	return &cr.(*corev1.Namespace).ObjectMeta
+}
+func (h namespaceHooks) reset( /* No implementation */ ) {}
+
+func (h *namespaceHooks) updateCr(req *common.HcoRequest, Client client.Client, exists runtime.Object, required runtime.Object) (bool, bool, error) {
+	namespace, ok1 := required.(*corev1.Namespace)
+	found, ok2 := exists.(*corev1.Namespace)
+	if !ok1 || !ok2 {
+		return false, false, errors.New("can't convert to Namespace")
+	}
+	needUpdate := false
+	for k, v := range namespace.Labels {
+		if found_v, hasKey := found.Labels[k]; hasKey {
+			if found_v != v {
+				found.Labels[k] = v
+				needUpdate = true
+			}
+		} else {
+			if found.Labels == nil {
+				found.Labels = make(map[string]string)
+			}
```

Comment:
Couldn't this be made outside the loop? Probably most of the time it would reduce the number of comparisons made.

Reply:
done